### PR TITLE
3.2.0

### PR DESCRIPTION
**Pull request by ultrafunkamsterdam**
added click_safe(), which could be used instead of click().  This will try to prevent detection when clicking on a link.

> added WebElement.children(self, tag=None, recursive=False)
  to easily get/find child nodes

> added example.py where i can point people at
  when asking silly questions
  (no, its actually quite cool, everyone should see it)

> some refactoring

> Solves #920

### CODE REVIEW

File: undetected_chromedriver/webelement.py [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+        """
+        returns direct child elements of current element
+        :param tag: str,  if supplied, returns <tag> nodes only
+        """
+        script = "return [... arguments[0].children]"
+        if tag:
+            script += ".filter( node => node.tagName === '%s')" % tag.upper()
+        if recursive:
+            return _recursive_children(self, tag)
+        return self._parent.execute_script(script, self)
```
Comment (hrt) [inline undetected_chromedriver/webelement.py:22]: Seems pretty expensive to call `execute_script` for every child of an element. Is it not easier to utilise js `getElementsByTagName`.

File: undetected_chromedriver/webelement.py [hunk 2]
```diff
@@ -35,3 +60,25 @@ def __repr__(self):
...
+    :param element: `WebElement` object.
+            find children below this <element>
+
+    :param tag: str = None.
+            if provided, return only <tag> elements. example: 'a', or 'img'
+    :param _results: do not use!
+    """
+    results = _results or set()
+    for element in element.children():
+        if tag:
```
Comment (hrt) [inline undetected_chromedriver/webelement.py:78]:
```py
if tag is None or element.tag_name == tag:
  results.add(element)
```
I think is less lines of code and improved readability

File: undetected_chromedriver/webelement.py [hunk 3]
```diff
@@ -35,3 +60,25 @@ def __repr__(self):
...
+            if element.tag_name == tag:
```
[the remark this answers is not in the document]
Comment (ultrafunkamsterdam) [reply undetected_chromedriver/webelement.py:79]: javascript behind it automatically covnerts to upper

File: undetected_chromedriver/webelement.py (hunk 1 above)
Comment (ultrafunkamsterdam) [reply undetected_chromedriver/webelement.py]: sure. And how do you think selenium get's their elements. all javascript calls....non issue

File: undetected_chromedriver/webelement.py (hunk 2 above)
Comment (ultrafunkamsterdam) [reply undetected_chromedriver/webelement.py:78]: if tag is None , as you suggest  and tag would be "" ,  than it would throw an Type Error.

**Pull request merged**